Helpers.gs

```diff
@@ -76,26 +76,32 @@ function fetchSourceCalendars(sourceCalendarURLs){
   for (var source of sourceCalendarURLs){
     var url = source[0].replace("webcal://", "https://");
     var colorId = source[1];
-
+    
     callWithBackoff(function() {
       var urlResponse = UrlFetchApp.fetch(url, { 'validateHttpsCertificates' : false, 'muteHttpExceptions' : true });
       if (urlResponse.getResponseCode() == 200){
-        var urlContent = RegExp("(BEGIN:VCALENDAR.*?END:VCALENDAR)", "s").exec(urlResponse.getContentText());
-        if(urlContent == null){
-          Logger.log("[ERROR] Incorrect ics/ical URL: " + url);
-          return;
-        }
-        else{
-          result.push([urlContent[0], colorId]);
-          return;
+        var icsContent = urlResponse.getContentText()
+        var icsRegex = RegExp("(BEGIN:VCALENDAR.*?END:VCALENDAR)", "s")
+        var urlContent = icsRegex.exec(icsContent);
+        if(urlContent == null) {
+          // Microsoft Outlook has a bug that sometimes results in incorrectly formatted ics files. This tries to fix that problem.
+          icsContent = icsContent.replaceAll(/(?<!END:VTIMEZONE[\r\n]+)BEGIN:VEVENT/g, 'END:VEVENT\nBEGIN:VEVENT') + 'END:VCALENDAR'
```

Comment:
Will Outlook always include VTIMEZONE components? If not, his will potentially create an invalid sequence
```
BEGIN:VCALENDAR
END:VEVENT
BEGIN:VEVENT
```

Reply:
@derekantrican My request was assigned a case and tracking ID number 2312060050002441 by the support agent, Vikash Kumar, but it was handled entirely by phone, email, and Quick Assist. I asked Vikash for documentation and updates on the case, but he expressed that no such public record would exist; the case would be handled exclusively internally by Microsoft.

@jonas0b1011001 that is a good question, and I don't know the answer. I don't have a lot of test cases on which to experiment. In the couple that I managed to create, there was only one instance of `END:VEVENT`, both times at the end of the ics file.

Reply:
@jonas0b1011001 I came up with a separate solution that (hopefully) avoids this issue. Would you be able to take a look? https://github.com/derekantrican/GAS-ICS-Sync/issues/388

Reply:
As i don't have much details about the bug i tried to create a more general fix, could you guys see if replacing L88 with this works:

```
//add END:VEVENT for every BEGIN:VEVENT that's missing it
const veventRegex = /BEGIN:VEVENT(?:(?!END:VEVENT).)*?(?=.BEGIN|.END:VCALENDAR|$)/sg;
icsContent = icsContent.replace(veventRegex, (match) => match + "\nEND:VEVENT");

//add END:VCALENDAR if missing
if (!icsContent.endsWith("END:VCALENDAR")) {
    icsContent += "\nEND:VCALENDAR";
}
```

Reply:
Teste @jonas0b1011001 and that works well.